### ISSUE TITLE
Terminology (subsequences and permutations)

### ISSUE THREAD

**Issue by lucasvreis** [Bodigrim/infinite-list#14]
I was confused when I saw these three functions on Hackage:
```hs
-- | Generate an infinite list of all subsequences of the argument.
subsequences :: Infinite a -> Infinite [a]

-- | Generate an infinite list of all non-empty subsequences of the argument.
subsequences1 :: Infinite a -> Infinite (NonEmpty a)

-- | Generate an infinite list of all permutations of the argument.
permutations :: Infinite a -> Infinite (Infinite a)
```
Due to cardinality, it's impossible to enumerate all subsequences or permutations of a given sequence. Of course, looking at the source code I realized they are enumerating (finite) sublists and finite permutations, so perhaps the docs (or even the names) could be adjusted to make this more clear for people used to the mathematical meanings 😅.

**Comment by Bodigrim** [comment]
Good point. Could you possibly raise a PR clarifying the docs? (I prefer to keep the names as is)

**Comment by Bodigrim** [comment]
Fixed in f302d91d8315eb50356e5ac123d5717d2fae3f67.
